week_2/pet.rb

```diff
@@ -110,6 +118,16 @@ def self.cost_to_feed(pets, days)
   #
   # Note - Order is not important
   def self.group_by_animal_type(pets)
-    raise NotImplementedError # TODO
+    temp=pets.group_by { |i| i.animal_type_id }
+      ans = {}
+      temp.each do |id,type|
+          names = []
+          type.each { |obj| 
+            names<<obj.name 
+          }
+          ans[id]=names
```

Comment:
[`map`](https://www.rubyguides.com/2018/10/ruby-map-method/) method can be used here